### DISCUSSION TITLE
Yup for validating JSON schema.

### DISCUSSION THREAD

**Discussion by b12k** [pactumjs/pactum#166] [Q&A] [answered]
👋🏻 Hello.

In the API docs I found `expectJsonSchema`. I wonder if there is a method which can accept external validators like [Yup](https://github.com/jquense/yup) validation.

There is a workaround `Yup => JsonSchema => expectJsonSchema`, where transformation to JsonSchema looks redundant.

Thank you in advance!

**Comment by ASaiAnudeep** [comment, marked as answer]
Hello @b12k, Welcome to PactumJS Discussions.

There are different ways we can bring our own custom assertion mechanisms. 

### [expect](https://pactumjs.github.io/api/assertions/expect.html)

**expect** method allows us to run some custom code to validate the response.

```js
const { spec } = require('pactum');
const { object, string, number } = require('yup');

const user_schema = object({
  name: string().required(),
  age: number().required().positive().integer()
});

await spec()
  .get('https://randomuser.me/api')
  .expect(ctx => {
    user_schema.validateSync(ctx.res.json);
  });
```

### [setJsonSchemaAdapter](https://pactumjs.github.io/api/settings/setJsonSchemaAdapter.html)

**setJsonSchemaAdapter** allows us to update the [inbuilt json schema adapter](https://github.com/pactumjs/pactum/blob/master/src/adapters/json.schema.js) to use a custom defined adapter.  Some additional work needs to be done to properly configure the new custom adapter.

**Reply by b12k** [reply]
Thanx a lot!
